`integration::test::package_json_basic` is flaky

https://github.com/denoland/deno/actions/runs/4267836955/jobs/7429836369

---- integration::test::package_json_basic stdout ----
deno_exe path /home/runner/work/deno/deno/target/release/deno
deno_exe args test
deno_exe cwd "/tmp/deno-cli-test-1677292511737-1038/package_json/basic"
output path /tmp/deno-cli-test-1677292511737-1038/package_json/basic/main.test.out
[test_util/src/lib.rs:2114] part = "/main.test.ts\n0\nrunning 1 test from "
[test_util/src/lib.rs:2114] i = 1
[test_util/src/lib.rs:2123] "exit false " = "exit false "
[test_util/src/lib.rs:2123] i = 1
OUTPUT
Download http://localhost:4545/npm/registry/@denotest/esm-basic
Download http://localhost:4545/npm/registry/@denotest/esm-basic/1.0.0.tgz
Check file:///tmp/deno-cli-test-1677292511737-1038/package_json/basic/main.test.ts
running 1 test from ./main.test.ts
should add ...
------- output -------
0
----- output end -----
should add ... ok (8ms)

ok | 1 passed | 0 failed (123ms)


OUTPUT
EXPECTED
Download http://localhost:4545/npm/registry/@denotest/esm-basic
Download http://localhost:4545/npm/registry/@denotest/esm-basic/1.0.0.tgz
Check file://[WILDCARD]/main.test.ts
0
running 1 test from [WILDCARD]main.test.ts
should add ... ok ([WILDCARD])

ok | 1 passed | 0 failed ([WILDCARD])


EXPECTED
thread 'integration::test::package_json_basic' panicked at 'pattern match failed', test_util/src/lib.rs:2074:7
stack backtrace:
   0:     0x560db510dbfa - std::backtrace_rs::backtrace::libunwind::trace::h34aec3ef6cd8ad7e
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x560db510dbfa - std::backtrace_rs::backtrace::trace_unsynchronized::h8035d38698d0f7a8
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x560db510dbfa - std::sys_common::backtrace::_print_fmt::hff968f6695a1ba22
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x560db510dbfa - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h7eea0efe77acf1ec
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x560db5136c2e - core::fmt::write::hc553f17407eb0b48
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/core/src/fmt/mod.rs:1208:17
   5:     0x560db5106795 - std::io::Write::write_fmt::he8b2346b6dea07fb
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/io/mod.rs:1682:15
   6:     0x560db510d9c5 - std::sys_common::backtrace::_print::h52d116aff3db4cd1
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/sys_common/backtrace.rs:47:5
   7:     0x560db510d9c5 - std::sys_common::backtrace::print::h9e7d2f98fb7af075
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/sys_common/backtrace.rs:34:9
   8:     0x560db510fe0f - std::panicking::default_hook::***closure***::hf74999dab2d0a95c
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:267:22
   9:     0x560db510facf - std::panicking::default_hook::hc11ca7d10c44c42f
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:283:9
  10:     0x560db41c41c7 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::hce89c32ffe00f6d8
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/alloc/src/boxed.rs:2032:9
  11:     0x560db41c41c7 - test::test_main::***closure***::heb0c177c7beb8e27
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/test/src/lib.rs:135:21
  12:     0x560db511059d - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::ha7dbb2d260f78172
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/alloc/src/boxed.rs:2032:9
  13:     0x560db511059d - std::panicking::rust_panic_with_hook::hdb4da1ae79c845a5
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:692:13
  14:     0x560db51102d2 - std::panicking::begin_panic_handler::***closure***::h02b5b35b126d5cf2
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:577:13
  15:     0x560db510e0ac - std::sys_common::backtrace::__rust_end_short_backtrace::h6c6853376cf416d1
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/sys_common/backtrace.rs:137:18
  16:     0x560db5110022 - rust_begin_unwind
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:575:5
  17:     0x560db5134ae3 - core::panicking::panic_fmt::hfd9e949092070b66
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/core/src/panicking.rs:64:14
  18:     0x560db5330627 - test_util::CheckOutputIntegrationTest::run::hf55403ebc2ce77f2
  19:     0x560db51dc35c - core::ops::function::FnOnce::call_once::hb55e079a10fe531a
  20:     0x560db41c964f - core::ops::function::FnOnce::call_once::h27fe4d61f5534446
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/core/src/ops/function.rs:507:5
  21:     0x560db41c964f - test::__rust_begin_short_backtrace::hca987cf6778f333e
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/test/src/lib.rs:657:18
  22:     0x560db4199b8c - test::run_test::***closure***::h83dee475f33cbcc1
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/test/src/lib.rs:648:30
  23:     0x560db4199b8c - core::ops::function::FnOnce::call_once***vtable.shim***::h35aac2a8dcc14b32
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/core/src/ops/function.rs:507:5
  24:     0x560db41c85da - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h91b1d309bad93834
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/alloc/src/boxed.rs:2000:9
  25:     0x560db41c85da - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h6923daf47f45e98d
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/core/src/panic/unwind_safe.rs:271:9
  26:     0x560db41c85da - std::panicking::try::do_call::h866512d118076d7e
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:483:40
  27:     0x560db41c85da - std::panicking::try::h7027435145a90584
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:447:19
  28:     0x560db41c85da - std::panic::catch_unwind::h268e92218421a872
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panic.rs:137:14
  29:     0x560db41c85da - test::run_test_in_process::h504c93dc6de288e7
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/test/src/lib.rs:680:27
  30:     0x560db41c85da - test::run_test::run_test_inner::***closure***::h4c2acc4b3be1f52c
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/test/src/lib.rs:574:39
  31:     0x560db41943f1 - test::run_test::run_test_inner::***closure***::h7f8e25639cc87b32
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/test/src/lib.rs:601:37
  32:     0x560db41943f1 - std::sys_common::backtrace::__rust_begin_short_backtrace::hce595d8354ce1b02
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/sys_common/backtrace.rs:121:18
  33:     0x560db4199dbb - std::thread::Builder::spawn_unchecked_::***closure***::***closure***::ha322fbc02fee5dd0
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/thread/mod.rs:550:17
  34:     0x560db4199dbb - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h9bdac2f448c3157f
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/core/src/panic/unwind_safe.rs:271:9
  35:     0x560db4199dbb - std::panicking::try::do_call::hfea270dc030e5c83
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:483:40
  36:     0x560db4199dbb - std::panicking::try::hef0ec4c778db4ba4
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:447:19
  37:     0x560db4199dbb - std::panic::catch_unwind::h675d2be7b6aff61f
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panic.rs:137:14
  38:     0x560db4199dbb - std::thread::Builder::spawn_unchecked_::***closure***::h1606be6d349fff0b
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/thread/mod.rs:549:30
  39:     0x560db4199dbb - core::ops::function::FnOnce::call_once***vtable.shim***::hddb03de14afe8277
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/core/src/ops/function.rs:507:5
  40:     0x560db51161c3 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::ha93239d92e7b610d
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/alloc/src/boxed.rs:2000:9
  41:     0x560db51161c3 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hebda57b0236de612
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/alloc/src/boxed.rs:2000:9
  42:     0x560db51161c3 - std::sys::unix::thread::Thread::new::thread_start::h1541b4390bbc26ca
                               at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/sys/unix/thread.rs:108:17
  43:     0x7f8dcf28a609 - start_thread
  44:     0x7f8dcf05e133 - clone
  45:                0x0 - <unknown>


failures:
    integration::test::package_json_basic

test result: FAILED. 1008 passed; 1 failed; 16 ignored; 0 measured; 0 filtered out; finished in 92.88s

error: test failed, to rerun pass `-p deno --test integration_tests`
Error: Process completed with exit code 101.

--- a/cli/tests/integration/bench_tests.rs
+++ b/cli/tests/integration/bench_tests.rs
@@ -220,7 +220,7 @@ fn file_protocol() {
 
 itest!(package_json_basic {
   args: "bench",
-  output: "package_json/basic/main.bench.out",
+  output: "package_json/basic/lib.bench.out",
   envs: env_vars_for_npm_tests(),
   http_server: true,
   cwd: Some("package_json/basic"),

--- a/cli/tests/integration/test_tests.rs
+++ b/cli/tests/integration/test_tests.rs
@@ -456,7 +456,7 @@ itest!(parallel_output {
 
 itest!(package_json_basic {
   args: "test",
-  output: "package_json/basic/main.test.out",
+  output: "package_json/basic/lib.test.out",
   envs: env_vars_for_npm_tests(),
   http_server: true,
   cwd: Some("package_json/basic"),

--- a/cli/tests/testdata/package_json/basic/fail_check.ts
+++ b/cli/tests/testdata/package_json/basic/fail_check.ts
@@ -1,3 +1,3 @@
-import { getValue } from "./main.ts";
+import { getValue } from "./lib.ts";
 
 const _test: string = getValue();

--- a/cli/tests/testdata/package_json/basic/lib.bench.out
+++ b/cli/tests/testdata/package_json/basic/lib.bench.out
@@ -1,11 +1,10 @@
 Download http://localhost:4545/npm/registry/@denotest/esm-basic
 Download http://localhost:4545/npm/registry/@denotest/esm-basic/1.0.0.tgz
-Check file:///[WILDCARD]/main.bench.ts
-0
+Check file:///[WILDCARD]/lib.bench.ts
 cpu: [WILDCARD]
 runtime: [WILDCARD]
 
-file:///[WILDCARD]/main.bench.ts
+file:///[WILDCARD]/lib.bench.ts
 [WILDCARD]
 -------------------------------------------------- -----------------------------
 should add  [WILDCARD]

--- a/cli/tests/testdata/package_json/basic/lib.bench.ts
+++ b/cli/tests/testdata/package_json/basic/lib.bench.ts
@@ -1,4 +1,4 @@
-import { add } from "./main.ts";
+import { add } from "./lib.ts";
 
 Deno.bench("should add", () => {
   if (add(1, 2) !== 3) {

--- a/cli/tests/testdata/package_json/basic/lib.test.out
+++ b/cli/tests/testdata/package_json/basic/lib.test.out
@@ -1,8 +1,7 @@
 Download http://localhost:4545/npm/registry/@denotest/esm-basic
 Download http://localhost:4545/npm/registry/@denotest/esm-basic/1.0.0.tgz
-Check file://[WILDCARD]/main.test.ts
-0
-running 1 test from [WILDCARD]main.test.ts
+Check file://[WILDCARD]/lib.test.ts
+running 1 test from [WILDCARD]lib.test.ts
 should add ... ok ([WILDCARD])
 
 ok | 1 passed | 0 failed ([WILDCARD])

--- a/cli/tests/testdata/package_json/basic/lib.test.ts
+++ b/cli/tests/testdata/package_json/basic/lib.test.ts
@@ -1,4 +1,4 @@
-import { add } from "./main.ts";
+import { add } from "./lib.ts";
 
 Deno.test("should add", () => {
   if (add(1, 2) !== 3) {

--- a/cli/tests/testdata/package_json/basic/lib.ts
+++ b/cli/tests/testdata/package_json/basic/lib.ts
@@ -1,0 +1,9 @@
+import * as test from "@denotest/esm-basic";
+
+export function add(a: number, b: number) {
+  return a + b;
+}
+
+export function getValue() {
+  return test.getValue();
+}

--- a/cli/tests/testdata/package_json/basic/main.info.out
+++ b/cli/tests/testdata/package_json/basic/main.info.out
@@ -1,7 +1,8 @@
 local: [WILDCARD]main.ts
 type: TypeScript
-dependencies: 1 unique
+dependencies: 2 unique
 size: [WILDCARD]
 
-file://[WILDCARD]/package_json/basic/main.ts ([WILDCARD])
-└── npm:@denotest/esm-basic@1.0.0 ([WILDCARD])
+file:///[WILDCARD]/main.ts (63B)
+└─┬ file:///[WILDCARD]/lib.ts (166B)
+  └── npm:@denotest/esm-basic@1.0.0 (345B)

--- a/cli/tests/testdata/package_json/basic/main.ts
+++ b/cli/tests/testdata/package_json/basic/main.ts
@@ -1,11 +1,3 @@
-import * as test from "@denotest/esm-basic";
+import { getValue } from "./lib.ts";
 
-console.log(test.getValue());
-
-export function add(a: number, b: number) {
-  return a + b;
-}
-
-export function getValue() {
-  return test.getValue();
-}
+console.log(getValue());

